libvast/vast/synopsis.hpp

```diff
@@ -68,24 +68,34 @@ class synopsis : public caf::ref_counted {
   /// Loads the contents for this slice from `source`.
   virtual caf::error deserialize(caf::deserializer& source) = 0;
 
+  /// @relates synopsis
+  friend inline bool operator==(const synopsis& x, const synopsis& y) {
+    return x.equals(y);
+  }
+
+  /// @relates synopsis
+  friend inline bool operator!=(const synopsis& x, const synopsis& y) {
+    return !(x == y);
+  }
+
 private:
   vast::type type_;
 };
 
-/// @relates synopsis
-inline bool operator==(const synopsis& x, const synopsis& y) {
-  return x.equals(y);
-}
-
-/// @relates synopsis
-inline bool operator!=(const synopsis& x, const synopsis& y) {
-  return !(x == y);
-}
-
 /// @relates synopsis
 caf::error inspect(caf::serializer& sink, synopsis_ptr& ptr);
 
 /// @relates synopsis
 caf::error inspect(caf::deserializer& source, synopsis_ptr& ptr);
 
+/// @relates synopsis
+static inline bool
+operator==(const synopsis_ptr& lhs, const synopsis_ptr& rhs) {
+  if (!lhs && !rhs)
+    return true;
+  if (!lhs || !rhs)
+    return false;
+  return *lhs == *rhs;
+}
+
```

Comment:
I don't like this operator, and think this should be rewritten for the `meta_index` specifically instead of doing a deep comparison for this pointer only.

Reply:
@tobim and I have decided on making the `operator==` of `meta_index` do this deep lookup instead.